### PR TITLE
4.0.8 custom logo

### PR DESCRIPTION
**Pull request by margud**

### CODE REVIEW

**Review by lukvai — APPROVED** [review]

**Review by aleksiuno — APPROVED** [review]

File: src/Config/Config.php [hunk 1]
```diff
@@ -234,6 +234,9 @@ class Config
     const MOLLIE_MODULE_UPGRADE_NOTICE_CLOSE_DATE = 'MOLLIE_MODULE_UPGRADE_NOTICE_CLOSE_DATE';
     const MODULE_UPGRADE_NOTICE_CLOSE_DURATION = 28;
 
+    const MOLLIE_SHOW_CUSTOM_LOGO = 'MOLLIE_SHOW_CUSTOM_LOGO';
+    const MOLLIE_CUSTOM_CREDIT_CARD_LOGO_NAME = 'customCreditCardLogo.jpg';
```
Comment (tomas862) [inline src/Config/Config.php:238]: you dont use this constant anymore

File: src/Provider/AbstractCustomLogoProvider.php [hunk 2]
```diff
@@ -0,0 +1,89 @@
...
+        return '';
+    }
+
+    /**
+     * @return string
+     */
+    public function getPathUri()
+    {
+        return '';
+    }
```
Comment (tomas862) [inline src/Provider/AbstractCustomLogoProvider.php:62]: why you need to implement this ?

File: src/Provider/CreditCardLogoProvider.php [hunk 3]
```diff
@@ -0,0 +1,89 @@
...
+            if ($this->logoExists()) {
+                return $this->getLogoPathUri();
+            }
+        }
+
+        $image = json_decode($methodObj->images_json, true);
+        $imageConfig = Configuration::get(Config::MOLLIE_IMAGES);
+
+        return ImageUtility::setOptionImage($image, $imageConfig);
+    }
```
Comment (tomas862) [inline src/Provider/CreditCardLogoProvider.php:88]: this does not exist in interface - so it should not exist at all.

File: src/Service/PaymentMethodService.php [hunk 4]
```diff
@@ -213,6 +220,11 @@ public function getMethodsForCheckout()
         foreach ($methods as $key => $method) {
             $image = json_decode($method['images_json'], true);
             $methods[$key]['image'] = $image;
+            if ($method['id_method'] === 'creditcard' && Configuration::get(Config::MOLLIE_SHOW_CUSTOM_LOGO)) {
+                if ($this->creditCardLogoProvider->logoExists()) {
+                    $methods[$key]['image']['custom_logo'] = $this->creditCardLogoProvider->getLogoPathUri();
+                }
+            }
```
Comment (tomas862) [inline src/Service/PaymentMethodService.php:227]: could you try not to nest if statements ?

**Pull request merged**